chore(build): Add support for drift detection of generated code

hermanschaaf opened the pull request:
This adds a new workflow that checks that generated code remains the same when go generate is run again. It currently needs to be specifically enabled by adding //check-for-changes to the top of the generator .hcl file.

cq-bot committed:
chore(build): Add support for drift detection of generated code

This adds a new workflow that checks that generated code remains the same when go generate is run again. It currently needs to be specifically enabled by adding //check-for-changes to the top of the generator .hcl file.

erezrokah reviewed (changes requested):
🔥 Same as 🚀  Same comment as https://github.com/cloudquery/cq-provider-gcp/pull/468#pullrequestreview-1064896387 regarding paths filters

hermanschaaf committed:
Refactor directory structure and fix drift

hermanschaaf committed:
update docs

hermanschaaf committed:
Fix path resolvers

hermanschaaf commented on resources/services/teams/teams.go:
```diff
@@ -370,7 +371,7 @@ func Teams() *schema.Table {
 						Name:        "team_id",
 						Description: "The id of the team",
-						Type:        schema.TypeBigInt,
+						Type:        schema.TypeString,
```
@roneli I don't know whether this should be `TypeBigInt` or `TypeString` -- the generator generates String, so I went with that for now.

hermanschaaf replied:
Oh nevermind, the tests failed. Guess it should be TypeBigInt after all

hermanschaaf commented on resources/services/teams/teams.go:
```diff
@@ -120,7 +121,7 @@ func Teams() *schema.Table {
 						Name:        "team_id",
 						Description: "The id of the name",
-						Type:        schema.TypeBigInt,
+						Type:        schema.TypeString,
```
@roneli I don't know whether this should be `TypeBigInt` or `TypeString` -- the generator generates String, so I went with that for now.

hermanschaaf committed:
rename to tool.go

hermanschaaf committed:
Fix team id type

hermanschaaf committed:
generate docs

hermanschaaf commented:
I have now also added the drift detection to all existing files and fixed any drift that had already occurred. To get the script to work also involved bringing the directory structure in line with other providers.

roneli reviewed (approved):
LGTM, one minor nit as we moved to packages feels maybe that billing/action_billing.go is redundant and we can do billing/actions etc' wdyt?

hermanschaaf commented:
@roneli I agree with you, however I think if we change the resource, that would also change the table name. Maybe let's do that in a separate PR if we're going to make a breaking change.
